### PR TITLE
Used interactor in handler and bookloader tests

### PR DESCRIPTION
**Pull request by StefanGajic**
https://trello.com/c/3BEEEYIp/111-use-the-interactor-in-tests-for-handler-and-bookloader

### CODE REVIEW

File: handler/handler.go [hunk 1]
```diff
@@ -5,34 +5,25 @@ import (
...
 }
 
-type openLibraryClient interface {
-	FetchBook(isbn string) (*openlibrarydto.Book, error)
+type bookInteractor interface {
+	FindAll() ([]*model.Book, error)
+	Create(bookRequest dto.CreateBookRequest) (*model.Book, error)
+	Update(book *model.Book, updateBookRequest dto.UpdateBookRequest) (*model.Book, error)
+	FindById(id int) (*model.Book, error)
+	Delete(book *model.Book) error
```
Comment (bokimilinkovic) [inline handler/handler.go]: why creating this interface here when we have struct Book in package interactor? 
https://github.com/sfqi/library/blob/chore/adding_interactor_to_handler/interactor/book.go

Comment (sfqi) [reply handler/handler.go]: @bokimilinkovic Are you objecting to the interface itself or to the fact that it's defined in this file (or package)?

Comment (bokimilinkovic) [reply handler/handler.go]: I am objecting to interface.. I think that we don't need this here, we have Book(from interactor package) with all methods attached to it

Comment (sfqi) [reply handler/handler.go]: Not sure why the new interface and `interactor.Book` would be in conflict - `interactor.Book` implements the interface. It's a dependency that we want to be able to mock.

File: middleware/bookloader.go [hunk 2]
```diff
@@ -3,15 +3,20 @@ package middleware
...
-	"github.com/sfqi/library/interactor"
 	"strconv"
 
 	"github.com/gorilla/mux"
+	"github.com/sfqi/library/domain/model"
+
 	"net/http"
 )
 
+type BookInteractor interface {
```
Comment (sfqi) [inline middleware/bookloader.go]: Make it unexported

File: handler/handler_test.go [hunk 3]
```diff
@@ -166,17 +162,21 @@ func TestUpdate(t *testing.T) {
...
-			Year:          2019,
-		}).Return(errors.New("error updating book"))
-		bookHandler.Db = db
+			Year:          2000,
+		}, dto.UpdateBookRequest{
+			Title: "test title",
+			Year:  2019,
+		},
+		).Return(nil, errors.New("error updating book"))
+		bookHandler.Book = interactor
```
Comment (sfqi) [inline handler/handler_test.go]: Could you also rename the field so that it's `bookHandler.Interactor = interactor`?

**Pull request merged**